File: 백준/Gold/1806. 부분합/부분합.py
import sys

input = sys.stdin.readline

n,m = map(int, input().split())
arr = [int(i) for i in input().split()]

def main():
    left = 0
    right = 0
    sumv = 0
    result = []
    while right <= len(arr):
        if sumv < m:
            if right >= len(arr):
                break
            sumv += arr[right]
            right += 1
            continue
            
        elif sumv >= m:
            result.append(right - left)
            sumv -= arr[left]
            left += 1
            
    result.sort()
    return result[0] if result else 0
            
    
print(main())

File: 백준/Gold/1806. 부분합/부분합.js
const fs=  require('fs');

const read = fs.readFileSync('dev/stdin').toString().trim();

let [nm, arr] = read.split("\n");

const [n, m] = nm.split(" ").map(Number);
arr = arr.split(" ").map(Number);

const main = () => {
    let left = 0;
    let right = 0;
    let sumv = 0;
    const result = [];
    while (right <= arr.length){
        if (sumv >= m){
            result.push(right - left)
            sumv -= arr[left++];
            continue
        }
        
        if (right >= arr.length) break;
        
        sumv += arr[right++];
    }
    
    result.sort((a,b) => a - b)
    return result.length ? result[0] : 0;
}

console.log(main())